WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/BaseUsernameChangerFullScreenDialogFragment.java

```diff
@@ -258,22 +258,30 @@ public void onUsernameSelected(String username) {
         mUsernameSelectedIndex = mUsernamesAdapter.getSelectedItem();
     }
 
-    protected String getUsernameOrSelected() {
+    private String getUsernameOrSelected() {
         return TextUtils.isEmpty(mUsernameSelected) ? mUsername : mUsernameSelected;
     }
 
-    protected void getUsernameSuggestions(String usernameQuery) {
+    public String getUsernameSelected() {
+        return mUsernameSelected;
+    }
+
+    private void getUsernameSuggestions(String usernameQuery) {
         showProgress(true);
 
         FetchUsernameSuggestionsPayload payload = new FetchUsernameSuggestionsPayload(usernameQuery);
         mDispatcher.dispatch(AccountActionBuilder.newFetchUsernameSuggestionsAction(payload));
     }
 
+    public UsernameChangerRecyclerViewAdapter getUsernamesAdapter() {
```

Comment:
It seems we use the adapter outside of this class just [here](https://github.com/jd-alexander/WordPress-Android/blob/4c23ba41d050bf28312a0880e06a8644019b087f/WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/UsernameChangerFullScreenDialogFragment.kt#L32 ) and [here](https://github.com/jd-alexander/WordPress-Android/blob/deed7c364ec973fde90aba6c6aa257b5ec5457b0/WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SettingsUsernameChangerFragment.kt#L68). Moreover, it seems the two places are almost identical.
How would you feel about removing this getter and keeping the adapter private just to the base class?

We could keep the following method in the base class.
```
@Override
    public boolean onConfirmClicked(FullScreenDialogController controller) {
        ActivityUtils.hideKeyboard(getActivity());

        if (mUsernamesAdapter != null && mUsernamesAdapter.mItems != null) {
            XYZ(controller, mUsernameSelected);
        } else {
            controller.dismiss();
        }

        return true;
    }
```

and introduce an abstract method just for the part which differs.
```
public abstract boolean XYZ(FullScreenDialogController controller, String userName);
```

Note: We could even remove the getter for `getUsernameSelected()`.

Wdyt?

Reply:
Fixed in https://github.com/wordpress-mobile/WordPress-Android/pull/10436/commits/f450a40ecfa27ef61d124ae91e5561bddf6e20e2